sql/parser/create.go

```diff
@@ -187,15 +187,21 @@ type IndexTableDef struct {
 
 func (node *IndexTableDef) String() string {
 	var buf bytes.Buffer
-	if node.Name != "" {
-		fmt.Fprintf(&buf, "CONSTRAINT %s ", node.Name)
-	}
 	if node.PrimaryKey {
```

Comment:
Yes, it is a bit ad-hoc, but it doesn't bother me that much. Yes, we could replace `node.PrimaryKey` and `node.Unique` with an enum, but I'm not sure that is any simpler given that this is only used in a single place in `sql/table.go` to create the `TableDescriptor`. 

I think separating `IndexTableDef` and `ConstraintTableDef` is premature right now because all of our constraints are currently indexes. At some point we may support `ForeignKeyTableDef`, but let's add that when we implement it (which may be never). Another way to look at this is that separating `IndexTableDef` and `ConstraintTableDef` provides zero additional functionality at this point while adding complexity to `sql.makeTableDesc`.

Would you really want to remove the ability to define indexes in `CREATE TABLE` and rely solely on `CREATE INDEX`? I know that is the postgres approach, but it seems so awkward.